### ISSUE TITLE
FocusTrapZone / isElementTabbable broken by div button

### ISSUE THREAD

**Issue by jhellesen** [OfficeDev/office-ui-fabric-react#7789]
FocusTrapZone fails to trap keyboard focus when the first or last element within the FocusTrapZone is a div, with role="button" and without a tab index.  This happens because the getLastTabbable and getFirstTabbable functions used by FocusTrapZone return the non-tabbable element and when the FocusTrapZone tries to focus on a non-tabbable element, the focus bounces out of the FocusTrapZone.

The function getLastTabbable and getFirstTabbable are incorrectly returning an a non-tabbable element because the function isElementTabbable does not properly checking the tab index.  In this case, the tabIndex for the element is null, but the final conditional in isElementTabbable is only checking if the tabIndex is -1, not null or undefined.  This allows isElementTabbable to return true when an element is not actually tabbable.

### Please provide a reproduction of the bug in a codepen:
 
https://codepen.io/anon/pen/rPOwBE

#### Actual behavior:
 
Keyboard focus leaves the FocusTrapZone when tabbing within the FocusTrapZone.
 
#### Expected behavior:
 
Expect keyboard focus to not leave the FocusTrapZone when tabbing within the FocusTrapZone.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: Normal

**Comment by leddie24** [comment]
Hey @jhellesen, can you verify that you posted the right codepen? I don't see any usage of `FocusTrapZone` within it.  Thanks!

**Comment by dzearing** [comment]
You can add `data-is-focusable="true"` to elements which should be considered focusable. Possible we could also alter that function to respect role="button".

**Comment by jhellesen** [comment]
> Hey @jhellesen, can you verify that you posted the right codepen? I don't see any usage of `FocusTrapZone` within it. Thanks!

Sorry - the Modal implicitly uses the FocusTrapZone.  Here is an more explicit example directly using the FocusTrapZone:

https://codepen.io/anon/pen/zeKMjJ
